Enhance Microsoft SSO user validation and error handling

DmitrySidorenko opened the pull request:
- Added error handling in `GetUserGroups` to return an empty list on failure.
- Introduced `ValidateUserSsoAccess` method in `ConnectService.cs` for improved user access validation.
- Updated `GetMicrosoftCodeBySsoId` to utilize the new validation method for user groups.

DmitrySidorenko committed:
Enhance Microsoft SSO user validation and error handling

- Added error handling in `GetUserGroups` to return an empty list on failure.
- Introduced `ValidateUserSsoAccess` method in `ConnectService.cs` for improved user access validation.
- Updated `GetMicrosoftCodeBySsoId` to utilize the new validation method for user groups.

AKercha1 commented on AiCoreApi/Services/ControllersServices/ConnectService.cs:
```diff
@@ -449,6 +437,71 @@ public async Task Logout(string idToken)
                 _logger.LogCritical("[{DateTime}][User Logout] Login: {Login}, Session id: {sessionId}", DateTime.UtcNow.ToString("g"), loginHistory.Login, sessionId);
         }
+
+        private async Task<SsoAccessValidationResult> ValidateUserSsoAccess(ExtendedTokenModel token)
```
Add Microsoft to the name as Microsoft is not the only provider

DmitrySidorenko replied:
Renamed

DmitrySidorenko commented on AiCoreApi/Services/ControllersServices/ConnectService.cs:
```diff
@@ -449,6 +437,71 @@ public async Task Logout(string idToken)
+
+    }
+
+    public class SsoAccessValidationResult
+    {
+        public bool IsValid { get; set; }
+        public List<ClientSsoModel> ValidConfigs { get; set; } = [];
+        public List<string> UserGroups { get; set; } = [];
+        public string? ErrorMessage { get; set; }
     }
 
```
Following the existing project pattern where service-specific Result classes 
are kept within their respective services

DmitrySidorenko commented on AiCoreApi/Common/SsoSources/MicrosoftSso.cs:
```diff
@@ -98,19 +98,26 @@ public string GetLoginRedirectUrl(string state)
+                    .Union(groupIds)
+                    .ToList();
+            }
+            catch
+            {
```
The empty list return is expected behavior, not an error that needs logging.

DmitrySidorenko committed:
Refactor: rename method for better clarity

- Rename ValidateUserMicrosoftSsoAccess to ValidateMicrosoftSsoAccess
- Keep security fixes (domain exposure removal)

DmitrySidorenko commented on AiCoreApi/Services/ControllersServices/ConnectService.cs:
```diff
@@ -449,6 +437,71 @@ public async Task Logout(string idToken)
+
+                return global.Count > 0
+                    ? new SsoAccessValidationResult { IsValid = true, ValidConfigs = global, UserGroups = [] }
+                    : new SsoAccessValidationResult { IsValid = false, ErrorMessage = $"Error: User domain '{userDomain}' is not allowed for Microsoft SSO." };
```
Removed domain exposure from error messages
